DEVOPS-452: Update package with python-poetry-template

SophieCurinier opened the pull request:
**DEVOPS-452 - apply latest version of python template repos to all repos**
- Use reusable github workflows
- Use MiraGeoscience/pre-commit-hooks package in `.pre-commit-hooks.yml`
- Update pre-commit-hooks version:
   - `ruff-pre-commit` : v0.5.5
   - `mirrors-mypy`: v1.10.1
   - `Mirageoscience/pre-commit-hooks`: v1.0.1
   - `codespell`: v2.3.0
   - `rstcheck`: v6.2.4

sebhmg commented on pyproject.toml:
```diff
@@ -94,11 +94,43 @@ h5py = ">=3.2.1, <4.0.0"  # from geoh5py
+    "B028",  # no-explicit-stacklevel for warnings.warn()
+    "E501",  # line-too-long - code is reformatted (do not care about comments and docstring)
+    "F401",  # unsused-import - covered by pycln
+    "RUF012", # TO DO
+    "UP038", # TO DO
+    "RUF015", # TO DO
```
"TODO" is a special comment marker highlighted in IDE.
Also used as the default keyword by some tools to extract and track TODOs (as in Jenkins)

```suggestion
    "RUF012",  # TODO
    "UP038",  # TODO
    "RUF015",  # TODO
```

SophieCurinier commented:
@sebhmg could you merge it? I don't have the right

sebhmg commented:
> @sebhmg could you merge it? I don't have the right

I have removed the branch protections and replaced by rulesets, with the updated list of required status check.
@SophieCurinier, do you have permission to merge now?

SophieCurinier commented:
> > @sebhmg could you merge it? I don't have the right
> 
> I have removed the branch protections and replaced by rulesets, with the updated list of required status check. @SophieCurinier, do you have permission to merge now?

No, I don't. It's propably because I'm not an admin on this repository.
